grails-app/domain/org/pih/warehouse/picklist/PicklistItem.groovy

```diff
@@ -40,7 +40,8 @@ class PicklistItem implements Serializable {
     InventoryItem inventoryItem
     Location binLocation
 
-    Integer quantity
+    Integer quantity // quantity = quantity picked
+    Integer quantityToPick
```

Comment:
you don't need to add the field to the toJson method?

Reply:
We need to add quantityPicked. The quantity property should be quantity to pick.

See OBKN https://github.com/openboxes/openboxes/blob/feature/obkn-integration/grails-app/domain/org/pih/warehouse/picklist/PicklistItem.groovy

Reply:
I'd like to bring in some of the other fields (like picker and datePicked) as well. 

![image](https://github.com/openboxes/openboxes/assets/1162935/34e10436-e6d4-4792-95e2-c40f0051c623)

Reply:
> you don't need to add the field to the toJson method?

Oh and to answer your question, Evan, we don't need to right now since in the PIH world quantity to pick (quantity) always equals quantity picked (quantityPicked). So essentially allocated and picked happen at exactly the same. This is not the ideal since there are important states and exceptions that occur between Allocated and Picked, but it's fine for PIH's default use case. At least for now. In addition, adding a state machine for the picking workflow would add overhead that would not likely not benefit for PIH until these events were important to track. That would change the minute PIH decided that it was important to allow short picks / overpicks or place a hold on a bin location when an item was not found in that location. 

Fwiw, the OBKN API (and the mobile app built on top) support a picking state machine that allows for a more complex picking workflow (including exceptions like partial pick, short picks, overpick, reallocation, etc). 

A typical state machine transition for a pick item might look like this.
```
Allocated -> To Pick -> Picking -> Picked
```